layers/meta-balena-fsl-arm/recipes-support/resin-init/resin-init-flasher-board/resin-init-flasher-board

```diff
@@ -0,0 +1,35 @@
+#!/bin/sh
+
+#
+# Re-use variation of boot update hook for initial flashing
+# Boot script is present inside the boot partition
+#
+
+set -o errexit
+
+# machine specific data
+uboot_file="imx-boot"
+uboot_block_size=1024
+uboot_seek_blocks=33
+device="/dev/mmcblk0boot0"
+
+update_files="uboot"
+
+for i in $update_files; do
+        current_update_file=$(eval echo \$${i}_file)
+        block_size=$(eval echo \$${i}_block_size)
+        seek_blocks=$(eval echo \$${i}_seek_blocks)
+
+        # calculate size and md5sum of the binary to update from the update bundle
+        update_size=$(ls -al /mnt/boot/$current_update_file | awk '{print $5}')
+        update_md5sum=$(md5sum /mnt/boot/$current_update_file | awk '{print $1'})
+
+        # calculate md5sum of the data already written to $device, using $update_size bytes
+        existing_md5sum=$(dd if=$device bs=1 count=$update_size status=none | md5sum | awk '{print $1}')
+
+        if [ ! "$existing_md5sum" = "$update_md5sum" ]; then
+                echo 0 > /sys/block/mmcblk0boot0/force_ro
+                dd if=/mnt/boot/$current_update_file of=$device conv=fdatasync seek=$seek_blocks bs=$block_size
+                echo 1 > /sys/block/mmcblk0boot0/force_ro
+        fi
+done
```

Comment:
Wondering if we need to go through the hassle of calculating existing bootloader data or just go ahead and do the `dd` without caring what was there before since we are flashing the board at this stage.